ext/curl/curl_url.stub.php

```diff
@@ -0,0 +1,240 @@
+
+    /**
+     * @var int
+     * @cname CURLU_URLDECODE
+     */
+    public const URLDECODE = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_URLENCODE
+     */
+    public const URLENCODE = UNKNOWN;
+
+#if LIBCURL_VERSION_NUM >= 0x074300 /* Available since 7.67.0 */
+    /**
+     * @var int
+     * @cname CURLU_NO_AUTHORITY
+     */
+    public const NO_AUTHORITY = UNKNOWN;
+#endif
+      
+#if LIBCURL_VERSION_NUM >= 0x074e00 /* Available since 7.78.0 */
+    /**
+     * @var int
+     * @cname CURLU_ALLOW_SPACE
+     */
+    public const ALLOW_SPACE = UNKNOWN;
+#endif
+
+    public function __construct(?string $url = null, int $flags = 0) {}
+
+    public function get(int $flags = 0): string {}
+    public function set(?string $url, int $flags = 0): CurlUrl {}
+
+    public function getHost(): ?string {}
+    public function setHost(?string $host): CurlUrl {}
+
+    public function getScheme(): ?string {}
+    public function setScheme(?string $scheme, int $flags = 0): CurlUrl {}
+
+    public function getPort(int $flags = 0): ?int {}
+    public function setPort(?int $port): CurlUrl {}
+
+    public function getPath(int $flags = 0): string {}
+    public function setPath(?string $scheme, int $flags = 0): CurlUrl {}
+
+    public function getQuery(int $flags = 0): ?string {}
+    public function setQuery(?string $query, int $flags = 0): CurlUrl {}
+
+    public function getFragment(int $flags = 0): ?string {}
+    public function setFragment(?string $fragment, int $flags = 0): CurlUrl {}
+
+    public function getUser(int $flags = 0): ?string {}
+    public function setUser(?string $user, int $flags = 0): CurlUrl {}
+
+    public function getPassword(int $flags = 0): ?string {}
+    public function setPassword(?string $password, int $flags = 0): CurlUrl {}
+
+    public function getOptions(int $flags = 0): ?string {}
+    public function setOptions(?string $options, int $flags = 0): CurlUrl {}
+
+#if LIBCURL_VERSION_NUM >= 0x074100 /* Available since 7.65.0 */
+    public function getZoneId(int $flags = 0): ?string {}
+    public function setZoneId(?string $zoneid, int $flags = 0): CurlUrl {}
+#endif
+ 
+    public function __toString(): string {}
+}
+
+final class CurlUrlException extends Exception
+{
+    /**
+     * @var int
+     * @cname CURLUE_BAD_PARTPOINTER
+     */
+    public const BAD_PARTPOINTER = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLUE_BAD_PORT_NUMBER
+     */
+    public const BAD_PORT_NUMBER = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLUE_MALFORMED_INPUT
+     */
+    public const MALFORMED_INPUT = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLUE_OUT_OF_MEMORY
+     */
+    public const OUT_OF_MEMORY = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLUE_UNSUPPORTED_SCHEME
+     */
+    public const UNSUPPORTED_SCHEME = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLUE_URLDECODE
+     */
+    public const URLDECODE = UNKNOWN;
```

Comment:
Does this mean that decoding was attempted and failed?

```suggestion
    public const URL_DECODING_FAILED = UNKNOWN;
```

Reply:
Yes it does. Thanks :-)